File: 2390 Removing Stars From a String/removing-stars-from-a-string.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string removeStars(string s) {
        string ret;
        int n = int(s.length());
        int u = 0;

        for (int i = 0; i < n; ++i) {
            if (s[i] == '*') ret.pop_back();
            else ret.push_back(s[i]);
        }
        
        return ret;
    }
};

int main() {
    Solution s;
    string st = "leet**cod*e";
    printf("%s\n", s.removeStars(st).c_str());
    st = "erase*****";
    printf("%s\n", s.removeStars(st).c_str());
}

File: 2390 Removing Stars From a String/Removing-Stars-From-a-String.py
class Solution:
    def removeStars(self, s: str) -> str:
        ret = ""
        n = len(s)
        n1 = 0

        for i in range(n):
            if s[i] != '*':
                ret += s[i]
                n1 += 1
            elif s[i] == '*':
                ret = ret[:n1 - 1]
                n1 -= 1

        return ret        

if __name__ == "__main__":
    s = Solution()
    print(s.removeStars("leet**cod*e"))
    print(s.removeStars("erase*****"))